### PR TITLE
Proposed 2.3.0-b4

### PR DESCRIPTION
**Pull request by intelliot**
## High Level Overview of Change

This is a beta for the 2.3.0 release.

This differs from #5141 in 2 ways:

1. It is rebased on the latest `develop`
2. It is a branch directly in this repo so that the repository secrets can be used by GitHub Actions

The base branch is `release`. [All releases (including betas)](https://github.com/XRPLF/rippled/blob/develop/CONTRIBUTING.md#before-you-start) go in `release`. This PR will be merged with `--ff-only` (not squashed or rebased, and not using the GitHub UI) to both `release` and `develop`.

### Context of Change

This beta includes the hotfix that was included in 2.2.3. There are also other changes; please review the commits associated with this PR to see what has changed since 2.3.0-b2.

### Type of Change

- [x] Release

### CODE REVIEW

**Comment by intelliot** [comment]
Per request from Nimesh, this PR will be in Draft until #5135 is approved and merged. (Note: that PR is currently dependent on #5140)

**Comment by intelliot** [comment]
Rebased on latest `develop`. Now ready for review.

File: RELEASENOTES.md
Comment (Bronek) [inline RELEASENOTES.md:9]: We will eventually need to add 2.3 section where we mention
- refactoring which affects projects with dependencies on this project
- removal of reporting mode
- removal of shards support

Not sure where we put the list of new amendments ?
featureNFTokenMintOffer
fixReducedOffersV2
fixEnforceNFTokenTrustline
fixInnerObjTemplate2
featureInvariantsV1_1
fixNFTokenPageLinks

This can wait until release candidate, so just mentioning with **no change requested**.

**Review by Bronek — APPROVED** [review]
Looking good, just one comment to remember before RC.

File: RELEASENOTES.md
Comment (intelliot) [reply RELEASENOTES.md:9]: Yes, I think the new amendments should be listed here under a new 2.3 section, in addition to being added on https://xrpl.org/resources/known-amendments which is the page I most frequently reference when it comes to amendments.

Those sound like breaking changes; I wonder if it's time for rippled 3.0 😀

Though now that there is Clio, I doubt anyone uses Reporting Mode anymore; I suppose 2.3 is fine.

**Pull request merged**
